design/api-design.md

```diff
@@ -0,0 +1,37 @@
+# API design
+
+## Requirements
+
+* API should be relatively easy to call (i.e., through a single GET/POST request)
+* It should be reasonably responsive by default (<5s)
+* API returns JSON object
+* If the schema is expected to change a lot, versioned API routes would be preferable.
+
+## Querying the API
+
+The API call includes all the results to be included in the map. This helps facilitate scheduled updates and reduces the need for repeat querying. An example query URL would be `api.epiverse.org/map` (versioned could look like `api.epiverse.org/v1/map`).
+
+## Response
+
+The response for the API is one JSON object containing an array of package names, the relevant metadata, and their respective embedding scores.
+
+```json
+{
+  "results": 2,
```

Comment:
It is common for API routes to return a JSON object that contains some metadata about the API call. I only included `results` at this time, but it could include some other metadata in the future (e.g., if we version it). 

Returning an array is technically not a JSON object either, which is partly why I did this too. It should not matter too much in any case as we can parse whatever comes in. It's more a convenience plus "convention" I've encountered.